domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -721,8 +721,29 @@ class ProfileManagementController @Inject constructor(
     return imageFile.absolutePath
   }
 
-  private fun onlyLetters(name: String): Boolean {
-    return name.matches(Regex("^[ A-Za-z]+\$"))
+  private fun nameAllowed(name: String): Boolean {
+    return (
+      notEmptyNoSpacesAndContainsLetters(name) &&
+        noNumbers(name) &&
+        noSymbols(name) &&
+        noRepeatedUseOfAllowedSymbols(name)
+      )
+  }
+
+  private fun notEmptyNoSpacesAndContainsLetters(name: String): Boolean {
+    return name.matches(Regex("^.[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'\\-]+\$"))
+  }
+
+  private fun noNumbers(name: String): Boolean {
+    return name.matches(Regex("^[^0-9]*\$"))
+  }
+
+  private fun noSymbols(name: String): Boolean {
+    return name.matches(Regex("^[^#*!@\$%^&()_+=\\\\|\\]\\[\":;?/><,`~{}]*\$"))
```

Comment:
Nit: you ought to define regex patterns as vals so that they don't need to be recompiled each time.

Also, why doesn't the letter check above cover the symbols case?

Reply:
> Also, why doesn't the letter check above cover the symbols case?
Since the regex pattern mentioned here https://github.com/oppia/oppia-android/pull/4081#discussion_r787941521 doesn't reject symbols, this function was added.

Reply:
I'm actually wondering if we could just check that it only contains letters and no repeated allowed symbols. Wouldn't the "else" case there be all other symbols (including whitespace, digits, and these symbols)?